openssl/src/cms.rs

```diff
@@ -202,11 +281,161 @@ impl CmsContentInfo {
     }
 }
 
+#[cfg(ossl110)]
+mod recipient_info {
+    use asn1::Asn1IntegerRef;
+    use stack::Stackable;
+    use x509::X509NameRef;
+    use foreign_types::Opaque;
+    use super::*;
+
+    impl CmsContentInfoRef {
+        /// Given that we're dealing with the `EnvelopedData`, returns the recipients of the message.
+        ///
+        /// OpenSSL documentation at [`CMS_get0_RecipientInfos`]
+        ///
+        /// [`CMS_get0_RecipientInfos`]: https://www.openssl.org/docs/manmaster/man3/CMS_RecipientInfo_decrypt.html
+        pub fn get_recipient_infos(&self) -> Result<&StackRef<CmsRecipientInfo>, ErrorStack> {
+            unsafe {
+                let recipient_infos = cvt_p(ffi::CMS_get0_RecipientInfos(self.as_ptr()))?;
+                Ok(StackRef::from_ptr(recipient_infos))
+            }
+        }
+    }
+
+    /// An owned version of [`CmsRecipientInfoRef`]
+    ///
+    /// [`CmsRecipientInfoRef`]:struct.CmsRecipientInfoRef.html
+    pub struct CmsRecipientInfo(*mut ffi::CMS_RecipientInfo);
+
+
+    impl ForeignType for CmsRecipientInfo {
+        type CType = ffi::CMS_RecipientInfo;
+        type Ref = CmsRecipientInfoRef;
+
+        #[inline]
+        unsafe fn from_ptr(ptr: *mut Self::CType) -> Self {
+            CmsRecipientInfo(ptr)
+        }
+
+        #[inline]
+        fn as_ptr(&self) -> *mut Self::CType {
+            self.0
+        }
+    }
+
+    /// Reference to a [`CMS_RecipientInfo`].
+    ///
+    /// [`CMS_RecipientInfo`]: https://www.openssl.org/docs/manmaster/man3/CMS_RecipientInfo_decrypt.html
+    pub struct CmsRecipientInfoRef(Opaque);
+
+    impl ForeignTypeRef for CmsRecipientInfoRef {
+        type CType = ffi::CMS_RecipientInfo;
+    }
+
+    impl Stackable for CmsRecipientInfo {
+        type StackType = ffi::stack_st_CMS_RecipientInfo;
+    }
+
+    /// The bindings to the openssl's type of `CMS_RecipientInfo` which can be extracted with
+    /// openssl's `CMS_RecipientInfo_ktri_get0_singer_id()` function.
+    pub enum RecipientInfo<'cms> {
+        Identifier,
+        Info {
+            issuer: &'cms X509NameRef,
+            serial_number: &'cms Asn1IntegerRef,
+        },
+    }
+
+    impl CmsRecipientInfoRef {
+        /// Implementation of [`CMS_RecipientInfo_ktri_get0_signer_id`].
+        ///
+        /// [`CMS_RecipientInfo_ktri_get0_signer_id`]: https://www.openssl.org/docs/manmaster/man3/CMS_RecipientInfo_decrypt.html
+        pub fn get_recipient_info<'cms>(&self) -> Result<RecipientInfo<'cms>, ErrorStack> {
```

Comment:
`'cms` isn't bound to an input here, so it's basically equivalent to `'static`. You'll need to do `&'cms self`.

Reply:
You're right, fixed.